File: BE-FindingJob/src/main/java/com/example/befindingjob/dto/employer/JobInfo.java
package com.example.befindingjob.dto.employer;

import com.example.befindingjob.dto.jobseeker.SkillInfo;
import com.example.befindingjob.entity.enumm.JobType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.HashSet;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class JobInfo {
    private int jobId;
    private String title;
    private CompanyInfo companyInfo;
    private String description;
    private String requirements;
    private String salary;
    private String location;
    private JobType jobType;
    private String experienceRequired;
    private String postingDate;
    private String expirationDate;
    private Set<SkillInfo> requiredSkills;
}

File: JobHub/app/src/main/java/com/example/jobhub/dto/employer/JobInfo.kt
package com.example.jobhub.dto.employer

import com.example.jobhub.dto.jobseeker.SkillInfo
import com.example.jobhub.entity.Application
import com.example.jobhub.entity.enumm.JobType

data class JobInfo(
    var jobId: Int? = null,
    var title: String = "",
    var companyInfo: CompanyInfo? = null,
    var description: String = "",
    var requirements: String = "",
    var salary: String = "",
    var location: String = "",
    var jobType: JobType? = null,
    var experienceRequired: String = "",
    var postingDate: String? = null,
    var expirationDate: String? = null,
    var requiredSkills: Set<SkillInfo> = emptySet(),
    var applications: Set<Application> = emptySet()
)
